### DISCUSSION TITLE
Marlin 2.1.2 - input shaping

### DISCUSSION THREAD

**Discussion by SargFowler** [mriscoc/Ender3V2S1#560] [Ideas]
Are there any plans to incorporate Marlin 2.1.2 which has input shaping?

**Comment by mriscoc** [comment]
Input Shaping was already implemented in the experimental version and will be available as a special configuration.

**Reply by Jag-1986** [reply]
Hello Mriscoc. :)

First of all! Thanks for making this awesome firmware! You are a legend! 👑 🥇 
 
I see that input shaping is now available in the new release.
What kind of sensor do you recommend?
And where do i connect it on the motherboard of the Ender 3 S1?
i have the F4 edition of the board.

**Reply by mriscoc** [reply]
Hello, Marlin IS needs to print a test pattern to adjust the parameters, currently does not support the use of accelerometers.
